compiler/rustc_lint/src/noop_method_call.rs

```diff
@@ -31,18 +31,48 @@ declare_lint! {
     /// calling `clone` on a `&T` where `T` does not implement clone, actually doesn't do anything
     /// as references are copy. This lint detects these calls and warns the user about them.
     pub NOOP_METHOD_CALL,
-    Allow,
+    Warn,
```

Comment:
I feel like this change and the associated changes to lib/compiler should be done in a separate PR.

Or at least the PR title and description need to be updated to reflect that this is also bumping NOOP_METHOD_CALL to warn.